lib/rouge/lexers/powershell.rb

```diff
@@ -652,28 +97,47 @@ class Powershell < Shell
         rule %r/\s+/, Text
         rule %r/\$?"/, Str::Double, :double_quotes
         rule %r/\$'/, Str::Single, :ansi_string
-
         rule %r/'/, Str::Single, :single_quotes
-
         rule %r/\*/, Keyword
-
         rule %r/;/, Text
         rule %r/[^=\*\s{}()$"'`<]+/, Text
         rule %r/\d+(?= |\Z)/, Num
         rule %r/</, Text
         mixin :interp
       end
 
+      state :hashtable do 
+        rule %r/\w/, Name::Variable
+        rule %r/}/, Operator, :pop!
+        mixin :root
+      end
+
+      state :multiline do
+        rule %r/\.(#{MULTILINEKEYWORDS})/i, Keyword::Pseudo
+        rule %r/[\w,\d,\s,\.,\-,\,:\/,{,},<,>"*]/, Comment
+        rule %r/#>/, Comment, :pop!
+        mixin :root
+      end
+
+      state :heredoc do
+        rule %r/"@/, Operator, :pop!
+        rule %r/[^$\n]+/, Str::Heredoc
+        rule %r/[$]/, Str::Heredoc
```

Comment:
Why exclude this from the rule above if you're just going to match it below?

Reply:
During testing that line would break things if it wasn't in there. I must have made a change later that made it unnecessary. If i remove 124 now, it just works. The heredoc portion is definitely going to require more work, and it would be nice to have some help there. 

```powershell
@" 
foo
$bar
@"
```
allows substitutions

```powerhsell
@'
foo
$bar
'@
````
would be a literal string.

Reply:
I think this should be working as intended now, please double check me.